### ISSUE TITLE
twisted.internet.address.UNIXAddress.__eq__ raises TypeError sometimes

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5653]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#5653|
|Type|defect|
|Created|2012-04-24 11:42:54Z|
|Branch|https://github.com/twisted/twisted/tree/none-unixaddress-comparison-5653|

Performing an equality comparison on `UNIXAddress(None)` will attempt to pass `None` to `os.path.samepath` sometimes.  This fails with a `TypeError`.

It's probably legitimate to have a `UNIXAddress(None)`, so this comparison should be fixed.

<details><summary>Searchable metadata</summary>

```
trac-id__5653 5653
type__defect defect
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_none_unixaddress_comparison_5653 branches/none-unixaddress-comparison-5653
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1335267774000000 1335267774000000
changetime__1335302384000000 1335302384000000
version__None None
owner__exarkun exarkun

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [34270]) Branching to 'none-unixaddress-comparison-5653'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

[Build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/none-unixaddress-comparison-5653)

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve set owner to @exarkun|
|-|-|

 * Some (existing I think) pyflakes:
```
twisted/internet/test/test_address.py:8: 'FilePath' imported but unused
twisted/internet/test/test_address.py:141: local variable 'address' is assigned to but never used
twisted/internet/test/test_address.py:171: local variable 'address' is assigned to but never used
twisted/internet/test/test_address.py:233: local variable 'address' is assigned to but never used
```
 * It'd be good to fix `__hash__` for this case as well.

+1 please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

Thanks!  Fixed those two points in r34275 and r34276.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set status to `closed`|
|-|-|

(In [34277]) Merge none-unixaddress-comparison-5653

Author: exarkun
Reviewer: therve
Fixes: [#5653](https://github.com/twisted/twisted/issues/5653)

Fix comparison and hashing of `twisted.internet.address.UNIXAddress` instances with `None`
as their "name".
